src/actors/Demon.gd

```diff
@@ -136,23 +138,32 @@ func take_damage(damage) -> void:
 	if health <= 0:
 		get_tree().change_scene("res://victory.tscn")
 
+
+func hit(damage, knockback = false, _owner = null) -> void:
+	if damage:
+		take_damage(damage)
+
+		if knockback:
+			knockback_target()
+
+func knockback_target():
+	position = position.move_toward(get_global_mouse_position(), 20)
+	knocked_back = true
+	yield(get_tree().create_timer(0.1), "timeout")
+	knocked_back = false
+
+
 func _on_Area2D_body_exited(body:Node):
-	if body.get_name() == "player":
+	if body.get_name().to_lower() == "player":
```

Comment:
There is no chance for user input to occur in the system, so calling `to_lower()` while safer, feels like a waste of additional resources/processing time, when we can just ensure its spelled correctly within the code itself.